### PR TITLE
[SuperEditor] Fix ActionTagComposingReaction looking for composing tag at selection base twice instead of at base and at extent.

### PR DESCRIPTION
**Pull request by KevinBrendel**
Without this fix, this could cause an exception, as the condition checks if `extent.nodePosition` is a `TextNodePosition` and then assumes that `base.nodePosition` is a `TextNodePosition`.

### CODE REVIEW

**Comment by matthew-carroll** [comment]
Can you please express this bug as a failing test and then show that this change fixes the bug?

**Comment by KevinBrendel** [comment]
Are there any existing tests related to action tags which I could use as a reference?

I currently don't have the setup to test it outside of our app, but essentially it can be triggered for example by:

1. Have one `ParagraphNode` with a cancelled ActionTag (careful not to trigger #2177)
2. Have a `HorizontalRuleNode`
3. Select upstream from the end of the `HorizontalRuleNode` into the `ParagraphNode`

It seems quite obvious to me that the code is not intended this way, as both the comment, as well as the condition indicate that extent is intended here. Base was already checked in the block before. (And it also fixes the exception, of course.)

https://github.com/user-attachments/assets/b6404371-c62a-4b97-b6ed-a69d1863a620

**Comment by matthew-carroll** [comment]
> Are there any existing tests related to action tags which I could use as a reference?

Yeah, we try to do our best to comprehensively test all features. In general, all PRs that modify behavior (non docs) are expected to include corresponding tests.

Here are the current action tag tests:

https://github.com/superlistapp/super_editor/blob/main/super_editor/test/super_editor/text_entry/tagging/action_tags_test.dart

**Comment by KevinBrendel** [comment]
@matthew-carroll Added a test which triggers the exception/demonstrates that the exception is no longer triggered with this fix.

File: super_editor/test/super_editor/text_entry/tagging/action_tags_test.dart [hunk 1]
```diff
@@ -651,6 +651,27 @@ void main() {
       });
     });
   });
+
+  group("selections >", () {
+    testWidgetsOnAllPlatforms("upstream cancelled action tag from block node", (tester) async {
```
Comment (matthew-carroll) [inline super_editor/test/super_editor/text_entry/tagging/action_tags_test.dart:656]: Can you come up with a better name for this test? As someone reading this who didn't work on the problem or the solution, I don't know what this string of words means...

File: super_editor/test/super_editor/text_entry/tagging/action_tags_test.dart [hunk 2]
```diff
@@ -651,6 +651,27 @@ void main() {
...
+      await tester.placeCaretInParagraph("1", 0);
+      await tester.typeImeText("/header ");
+
+      // Place cursor at the end of the horizontal rule/block node
+      await tester.pressDownArrow();
+      await tester.pressRightArrow();
+
+      // Select upstream towards the cancelled action tag
+      await tester.pressShiftLeftArrow();
+      await tester.pressShiftUpArrow();
```
Comment (matthew-carroll) [inline super_editor/test/super_editor/text_entry/tagging/action_tags_test.dart:672]: If possible, we'd like to formulate tests in a way that we can naturally declare expectations that are met in the test.

If there's no way to formulate this test with an expectation that makes sense, then at a minimum we need a comment at the bottom of this test explaining that reaching the comment without throwing an exception means the test passes. Please also briefly describe the bug that lead to this test, and mention the issue ticket number in the comment.

File: super_editor/test/super_editor/text_entry/tagging/action_tags_test.dart (hunk 1 above)
Comment (KevinBrendel) [reply super_editor/test/super_editor/text_entry/tagging/action_tags_test.dart]: Added a longer name to express it more explicitly.

File: super_editor/test/super_editor/text_entry/tagging/action_tags_test.dart (hunk 2 above)
Comment (KevinBrendel) [reply super_editor/test/super_editor/text_entry/tagging/action_tags_test.dart]: Added expectLater with returnsNormally. This does not appear to change the behavior of the test, but makes it more explicit. Also added comment and link.

**Comment by KevinBrendel** [comment]
Not sure why the netlify deploys are failing now. Is it because this PR's base is now outdated vs the main branch?

**Comment by matthew-carroll** [comment]
I tried to edit this PR directly so you wouldn't need to push another change, but it looks like I can't do that.

Please replace the test description with this:

```
"can find tag that surrounds the extent position when the selection is expanded"
```

Please replace the description at the bottom of the test with this:

```
// If we reach the end without exception, then ActionTagComposingReaction did not blow up due to the base or extent
// position, and type of content at those positions.
//
// Original bug: https://github.com/superlistapp/super_editor/pull/2201
```

**Comment by KevinBrendel** [comment]
@matthew-carroll Done.

**Review by matthew-carroll — APPROVED** [review]
LGTM

**Comment by matthew-carroll** [comment]
@angelosilvestre can you cherry pick this?

**Pull request merged**